src/Graph.ts

```diff
@@ -453,7 +444,15 @@ export default class Graph {
 
 			module.dependencies.forEach(depModule => {
 				if (!depModule.isExternal) {
-					visit(<Module>depModule, seen);
+					if (depModule.id in parents) {
+						if (!allSeen[depModule.id]) {
+							this.warnCycle(depModule.id, { ...parents, [depModule.id]: module.id });
```

Comment:
The parents expansion shouldn't be necessary now that this is inline I believe?

Reply:
We may not alter `parents` here because that would break for other cycles involving the same modules. We could do `Object.assign(Object.create(parents}, { [depModule.id]: module.id })`, that's a lot simpler to create but takes more time on lookup.

Reply:
But parents isn't changed while `warnCycle` runs synchronously right?

Reply:
No, but we may not set `parents[depModule.id]` either. We could pass `module.id` separately, though.

Reply:
Yes that sounds sensible.

Reply:
Did that. Looks better :)